invisible_cities/reco/icaro_components.py

```diff
@@ -0,0 +1,219 @@
+        If the selection efficiency is out of this interval
+        the map production will abort/just warn, depending on "strictness".
+        Defaults to [0, 1].
+    strictness: Strictness (Optional)
+        If 'warning', function returns a False if the criteria
+        is not matched. If 'stop_proccess' it raises an exception.
+        Defaults to `raise_error`.
+    Returns
+    ----------
+        A mask corresponding to the selected events.
+    """
+    input_mask = input_mask if input_mask is not None else [True] * len(dst)
+    mask             = np.zeros_like(input_mask)
+    mask[input_mask] = dst.loc[input_mask, column.value] == 1
+
+    nevts_after      = dst[mask]      .event.nunique()
+    nevts_before     = dst[input_mask].event.nunique()
+    eff              = nevts_after / nevts_before
+    all_in_range(data         = np.array(eff)  ,
+                 minval       = eff_interval[0],
+                 maxval       = eff_interval[1],
+                 display_name = column.value   ,
+                 strictness   = strictness     ,
+                 right_closed = True)
+
+    return mask
+
+
+def select_band_and_check(dst         : pd.DataFrame                                              ,
+                          boot_map    : ASectorMap                                                ,
+                          norm_strat  : Optional[NormStrategy]                  = NormStrategy.max,
+                          input_mask  : Optional[np.ndarray]                    = None            ,
+                          range_DT    : Optional[Tuple[np.ndarray, np.ndarray]] = (10, 1300)      ,
+                          range_E     : Optional[Tuple[np.ndarray, np.ndarray]] = (10.0e+3,14e+3) ,
+                          nsigma_sel  : Optional[float]                         = 3.5             ,
+                          eff_interval: Optional[Tuple[float, float]]           = [0,1]           ,
+                          strictness  : Optional[Strictness]                    = Strictness.raise_error
+                          )->np.array:
+    """
+    This function returns a selection of the events that
+    are inside the Kr E vz Z band, and checks
+    if the selection efficiency is correct.
+
+    Parameters
+    ----------
+    dst : pd.DataFrame
+        Krypton dataframe.
+    boot_map: str
+        Name of bootstrap map file.
+    norm_strat: norm_strategy
+        Provides the desired normalization to be used.
+        Defaults to `max`.
+    mask_input: np.array (Optional)
+        Mask of the previous selection cut.
+        Defaults to None.
+    range_DT: Tuple[np.array, np.array] (Optional)
+        Range in DT-axis
+        Defaults to (10, 300).
+    range_E: Tuple[np.array, np.array] (Optional)
+        Range in Energy-axis.
+        Defaults to (10e3, 14e3).
+    nsigma_sel: float (Optional)
+        Number of sigmas to set the band width.
+        Defaults to 3.5.
+    eff_interval (Optional)
```

Comment:
```suggestion
    eff_interval: (float, float), optional
```

```suggestion
    eff_interval: Tuple[float, float], optional
```
Please keep on of these formats everywhere to make the style of the documentation consistent. We are heavily inspired by numpy's documentation, so in case of doubt, that is the reference.